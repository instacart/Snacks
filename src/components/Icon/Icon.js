import React from 'react'
import PropTypes from 'prop-types'
import hexValues from './hexValues'

const baseStyles = {
  fontSize: '16px',
  position: 'relative',
  fontFamily: 'ic-icons',
  fontStyle: 'normal',
  fontWeight: 'normal',
  fontVariant: 'normal',
  textTransform: 'none',
  speak: 'none',
  lineHeight: '1',
  fontSmoothing: 'antialiased',
  osxFontSmoothing: 'grayscale',
}

const getIcon = ({ name, code }) => {
  const iconCode = !code ? hexValues[name] : code
  const codePoint = parseInt(iconCode, 16)
  return String.fromCodePoint(codePoint)
}

const Icon = props => {
  const { className, style, onClick } = props
  const icon = getIcon(props)
  return (
    <i className={className} css={[baseStyles, style]} aria-hidden onClick={onClick}>
      {icon}
    </i>
  )
}

Icon.propTypes = {
  className: PropTypes.string,
  /**
   * String name of icon - ex 'cart'
   */
  name: PropTypes.oneOf(Object.keys(hexValues)),
  /** Hexcode of desired icon from ic-icons */
  code: PropTypes.string,
  /**
   * Optional style overrides merged into emotion css
   *
   * @deprecated
   * This prop exists for backwards compatibility and will be
   * removed in a future version
   */
  style: PropTypes.object,
  /** Callback function called after button click
   * @param {SyntheticEvent} event The react `SyntheticEvent`
   */
  onClick: PropTypes.func,
}

export default Icon
