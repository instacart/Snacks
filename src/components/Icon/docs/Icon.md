Icon example:
```jsx
<div>
  <Icon name={'cart'} />
  <Icon code={'e08e'} />
  <Icon
    name={'likeFilled'}
    style={{
      backgroundColor: '#43B02A',
      color: '#fff',
      borderRadius: '50%',
      padding: '5px'
    }}
  />
</div>
```

See full list of available icons under the Icons section.
