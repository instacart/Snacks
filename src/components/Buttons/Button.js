import React from 'react'
import PropTypes from 'prop-types'
import Radium from 'radium'
import Icon from '../Icon/Icon'
import withTheme from '../../styles/themer/withTheme'
import { themePropTypes } from '../../styles/themer/utils'
import { spacing, colors } from '../../styles'
import { darken } from '../../utils'

const noop = () => {} // eslint-disable-line no-empty-function

const baseStyles = {
  touchAction: 'manipulation',
  cursor: 'pointer',
  border: '1px solid transparent',
  borderTopRightRadius: 4,
  borderBottomRightRadius: 4,
  borderBottomLeftRadius: 4,
  borderTopLeftRadius: 4,
  fontWeight: 600,
  whiteSpace: 'nowrap',
  userSelect: 'none',
  WebkitFontSmoothing: 'antialiased',

  // Reset unusual Firefox-on-Android default style;
  // see https://github.com/necolas/normalize.css/issues/214
  backgroundImage: 'none',

  // Takes care of the heavier default font-weights in firefox
  MozOsxFontSmoothing: 'grayscale',

  // Ensures any icons stay vertically centered
  display: 'inline-flex',
  alignItems: 'center'
}

const sizeStyles = {
  tiny: {
    ...spacing.PADDING_X_XS,
    height: '24px',
    fontSize: '11px',
    textTransform: 'uppercase'
  },

  small: {
    ...spacing.PADDING_X_XS,
    fontSize: '14px',
    height: '32px'
  },

  standard: {
    ...spacing.PADDING_X_SM,
    fontSize: '16px',
    height: '40px'
  },

  large: {
    ...spacing.PADDING_X_MD,
    fontSize: '18px',
    height: '48px'
  }
}

const linkStyles = {
  display: 'flex',
  alignItems: 'center',
  boxSizing: 'border-box'
}

const getSnacksStyles = props => {
  const { action, actionHover, primaryBackground } = props.snacksTheme.colors
  const actionActive = darken(actionHover, 3)

  const disabled = {
    backgroundColor: colors.GRAY_74,
    color: colors.WHITE,
    cursor: 'not-allowed'
  }

  const activeAndFocus = {
    primary: {
      base: {
        backgroundColor: actionActive
      },
      inverted: {
        backgroundColor: primaryBackground,
        opacity: '0.8'
      }
    },
    secondary: {
      base: {
        color: actionActive,
        border: `1px solid ${actionActive}`
      },
      inverted: {
        color: primaryBackground,
        border: `1px solid ${primaryBackground}`,
        opacity: '0.8'
      }
    },
    flat: {
      color: actionActive,
    },
    coupon: {
      border: `1px dashed ${colors.RED_700}`,
      color: colors.RED_700
    }
  }

  return {
    primary: {
      base: {
        backgroundColor: action,
        color: primaryBackground,

        ':hover': {
          backgroundColor: actionHover
        },
        ':active': activeAndFocus.primary.base,
        ':focus': activeAndFocus.primary.base
      },

      inverted: {
        backgroundColor: primaryBackground,
        border: `1px solid ${action}`,
        color: action,

        ':hover': {
          backgroundColor: primaryBackground,
          opacity: '0.9'
        },
        ':active': activeAndFocus.primary.inverted,
        ':focus': activeAndFocus.primary.inverted
      },

      disabled
    },

    secondary: {
      base: {
        backgroundColor: 'transparent',
        color: action,
        border: `1px solid ${action}`,

        ':hover': {
          color: actionHover,
          border: `1px solid ${actionHover}`
        },
        ':active': activeAndFocus.secondary.base,
        ':focus': activeAndFocus.secondary.base
      },

      inverted: {
        color: primaryBackground,
        border: `1px solid ${primaryBackground}`,

        ':hover': {
          color: primaryBackground,
          border: `1px solid ${primaryBackground}`,
          opacity: '0.9'
        },
        ':active': activeAndFocus.secondary.inverted,
        ':focus': activeAndFocus.secondary.inverted
      },

      disabled
    },

    flat: {
      base: {
        backgroundColor: 'transparent',
        color: action,

        ':hover': {
          color: actionHover,
        },
        ':active': activeAndFocus.flat,
        ':focus': activeAndFocus.flat
      },

      disabled: {
        backgroundColor: 'transparent',
        color: colors.GRAY_74,
        cursor: 'not-allowed'
      }
    },

    coupon: {
      base: {
        backgroundColor: colors.WHITE,
        color: colors.RED_500,
        border: `1px dashed ${colors.RED_500}`,

        ':hover': {
          border: `1px dashed ${colors.RED_600}`,
          color: colors.RED_600
        },
        ':active': activeAndFocus.coupon,
        ':focus': activeAndFocus.coupon
      },

      disabled
    }
  }
}

const Button = props => {
  const snacksStyles = getSnacksStyles(props)

  const ElementType = props.href ? 'a' : 'button'

  const finalProps = {
    disabled: props.disabled,
    tabIndex: props.tabIndex,
    type: props.type,
    style: [
      baseStyles,
      sizeStyles[props.size],
      snacksStyles[props.snacksStyle][props.disabled ? 'disabled' : 'base'],
      props.inverted && snacksStyles[props.snacksStyle].inverted,
      ElementType === 'a' && linkStyles,
      props.style
    ],
    onClick: e => {
      if (props.disabled) {
        e.preventDefault()
        return
      }

      props.onClick(e, props)
    },
    onMouseDown: e => {
      if (props.disabled) {
        e.preventDefault()
        return
      }

      props.onMouseDown(e, props)
    },
    ...props.elementAttributes
  }
  if (props.href) {
    finalProps.href = props.href
  }

  const icon =
    typeof props.icon === 'string' ? <Icon name={props.icon} /> : props.icon

  if (icon && props.iconPosition === 'left') {
    return (
      <ElementType {...finalProps}>
        {icon}
        <span style={{ ...spacing.MARGIN_RIGHT_XS }} />
        {props.children}
      </ElementType>
    )
  }

  if (icon && props.iconPosition === 'right') {
    return (
      <ElementType {...finalProps}>
        {props.children}
        <span style={{ ...spacing.MARGIN_LEFT_XS }} />
        {icon}
      </ElementType>
    )
  }

  return <ElementType {...finalProps}>{props.children}</ElementType>
}

Button.propTypes = {
  /** Size of the button. */
  size: PropTypes.oneOf(['tiny', 'small', 'standard', 'large']),

  /** Snacks button variants. */
  snacksStyle: PropTypes.oneOf(['primary', 'secondary', 'flat', 'coupon']),

  /** Optional style overrides. */
  style: PropTypes.object,

  /** Whether or not the button is disabled. */
  disabled: PropTypes.bool,

  /** Sets the HTML type attribute on the element. */
  type: PropTypes.oneOf(['button', 'reset', 'submit']),

  /**
    An optional icon. Can be a an icon name
    or a Snacks `Icon` component.
  */
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),

  /** Where the icon is positioned relative to primary text content. */
  iconPosition: PropTypes.oneOf(['left', 'right']),

  /** Callback that fires when the button is clicked. */
  onClick: PropTypes.func,

  /** Callback that fires on mouse down. */
  onMouseDown: PropTypes.func,

  /** Reverses colors. Use for rendering buttons on dark backgrounds. */
  inverted: PropTypes.bool,

  /** Sets the HTML `tabindex` attribute. */
  tabIndex: PropTypes.number,

  /**
    If href is provided, `Button` will render as an anchor
    tag with the supplied href value.
  */
  href: PropTypes.string,

  /** Text content for the button. */
  children: PropTypes.node,

  /** Any additonal props to add to the element (e.g. data attributes). */
  elementAttributes: PropTypes.object,

  /** Snacks theme attributes provided by `Themer` */
  snacksTheme: themePropTypes
}

Button.defaultProps = {
  size: 'standard',
  snacksStyle: 'primary',
  disabled: false,
  type: 'button',
  iconPosition: 'left',
  onClick: noop,
  onMouseDown: noop,
  inverted: false,
  elementAttributes: {}
}

export default withTheme(Radium(Button))
