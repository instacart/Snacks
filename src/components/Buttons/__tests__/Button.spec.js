import React from 'react'
import renderer from 'react-test-renderer'
import { StyleRoot } from 'radium'
import { mount } from 'enzyme'
import { spy } from 'sinon'
import Icon from '../../Icon/Icon'
import Button from '../Button'

describe('Button', () => {
  it('renders all sizes correctly', () => {
    const testCases = [
      { snacksStyle: 'primary', size: 'tiny' },
      { snacksStyle: 'primary', size: 'small' },
      { snacksStyle: 'primary', size: 'standard' },
      { snacksStyle: 'primary', size: 'large' },
    ]

    testCases.forEach(props => {
      const tree = renderer
        .create(
          <StyleRoot>
            <Button {...props}>Primary Button</Button>
          </StyleRoot>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  it('renders all snacks button variants correctly', () => {
    const testCases = [
      { snacksStyle: 'primary', size: 'standard' },
      { snacksStyle: 'secondary', size: 'standard' },
      { snacksStyle: 'flat', size: 'standard' },
      { snacksStyle: 'coupon', size: 'standard' },
    ]

    testCases.forEach(props => {
      const tree = renderer
        .create(
          <StyleRoot>
            <Button {...props}>Hi</Button>
          </StyleRoot>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  it('renders with inverted colors', () => {
    const tree = renderer
      .create(
        <StyleRoot>
          <Button inverted snacksType="primary" size="small">
            Hi
          </Button>
        </StyleRoot>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders icons correctly', () => {
    const testCases = [
      { icon: 'cart' },
      { icon: 'cart', iconPosition: 'right' },
      { icon: <Icon name="cart" /> },
      { icon: <Icon name="cart" />, iconPosition: 'right' },
    ]

    testCases.forEach(props => {
      const tree = renderer
        .create(
          <StyleRoot>
            <Button {...props}>Hi</Button>
          </StyleRoot>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  it('applies the elementAttributes prop correctly', () => {
    const tree = renderer
      .create(
        <StyleRoot>
          <Button elementAttributes={{ 'aria-label': 'foo' }} nacksType="primary">
            Hi
          </Button>
        </StyleRoot>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('fires the onClick prop', () => {
    const onClick = spy()
    const wrapper = mount(
      <StyleRoot>
        <div>
          <Button snacksType="primary" onClick={onClick}>
            Hi
          </Button>
        </div>
      </StyleRoot>
    )

    wrapper.find('button').simulate('click')

    expect(onClick.calledOnce).toBe(true)
  })

  it('renders correctly when disabled', () => {
    const tree = renderer
      .create(
        <StyleRoot>
          <Button disabled snacksType="primary">
            Hi
          </Button>
        </StyleRoot>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('does not fire the onClick prop when disabled', () => {
    const onClick = spy()
    const wrapper = mount(
      <StyleRoot>
        <Button disabled snacksType="primary">
          Hi
        </Button>
      </StyleRoot>
    )

    wrapper.find('button').simulate('click')

    expect(onClick.calledOnce).toBe(false)
  })

  it('can render as a link if an href is provided', () => {
    const tree = renderer.create(
      <StyleRoot>
        <Button href="/carrot" snacksType="primary">
          Hi
        </Button>
      </StyleRoot>
    )
    expect(tree).toMatchSnapshot()
  })
})
