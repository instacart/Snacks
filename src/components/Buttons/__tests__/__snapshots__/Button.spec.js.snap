// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button applies the elementAttributes prop correctly 1`] = `
<div
  data-radium={true}
>
  <button
    aria-label="foo"
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Hi
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button can render as a link if an href is provided 1`] = `
<div
  data-radium={true}
>
  <a
    data-radium={true}
    disabled={false}
    href="/carrot"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "boxSizing": "border-box",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Hi
  </a>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders all sizes correctly 1`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "11px",
        "fontWeight": 600,
        "height": "24px",
        "paddingLeft": 8,
        "paddingRight": 8,
        "textTransform": "uppercase",
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Primary Button
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders all sizes correctly 2`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "14px",
        "fontWeight": 600,
        "height": "32px",
        "paddingLeft": 8,
        "paddingRight": 8,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Primary Button
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders all sizes correctly 3`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Primary Button
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders all sizes correctly 4`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "18px",
        "fontWeight": 600,
        "height": "48px",
        "paddingLeft": 24,
        "paddingRight": 24,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Primary Button
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders all snacks button variants correctly 1`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Hi
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders all snacks button variants correctly 2`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "transparent",
        "backgroundImage": "none",
        "border": "1px solid #43B02A",
        "borderRadius": "4px",
        "color": "#43B02A",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Hi
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders all snacks button variants correctly 3`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "backgroundImage": "none",
        "border": "1px dashed #E0E0E0",
        "borderRadius": "4px",
        "color": "#E6003D",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Hi
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders correctly when disabled 1`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#BDBDBD",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "not-allowed",
        "display": "inline-flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Hi
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders icons correctly 1`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    <i
      aria-hidden={true}
      data-radium={true}
      onClick={undefined}
      style={
        Object {
          "fontFamily": "ic-icons",
          "fontSize": "16px",
          "fontSmoothing": "antialiased",
          "fontStyle": "normal",
          "fontVariant": "normal",
          "fontWeight": "normal",
          "lineHeight": "1",
          "osxFontSmoothing": "grayscale",
          "position": "relative",
          "speak": "none",
          "textTransform": "none",
        }
      }
    >
      
    </i>
    <span
      data-radium={true}
      style={
        Object {
          "marginRight": 8,
        }
      }
    />
    Hi
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders icons correctly 2`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Hi
    <span
      data-radium={true}
      style={
        Object {
          "marginLeft": 8,
        }
      }
    />
    <i
      aria-hidden={true}
      data-radium={true}
      onClick={undefined}
      style={
        Object {
          "fontFamily": "ic-icons",
          "fontSize": "16px",
          "fontSmoothing": "antialiased",
          "fontStyle": "normal",
          "fontVariant": "normal",
          "fontWeight": "normal",
          "lineHeight": "1",
          "osxFontSmoothing": "grayscale",
          "position": "relative",
          "speak": "none",
          "textTransform": "none",
        }
      }
    >
      
    </i>
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders icons correctly 3`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    <i
      aria-hidden={true}
      data-radium={true}
      onClick={undefined}
      style={
        Object {
          "fontFamily": "ic-icons",
          "fontSize": "16px",
          "fontSmoothing": "antialiased",
          "fontStyle": "normal",
          "fontVariant": "normal",
          "fontWeight": "normal",
          "lineHeight": "1",
          "osxFontSmoothing": "grayscale",
          "position": "relative",
          "speak": "none",
          "textTransform": "none",
        }
      }
    >
      
    </i>
    <span
      data-radium={true}
      style={
        Object {
          "marginRight": 8,
        }
      }
    />
    Hi
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders icons correctly 4`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#43B02A",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "color": "#FFFFFF",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "16px",
        "fontWeight": 600,
        "height": "40px",
        "paddingLeft": 16,
        "paddingRight": 16,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Hi
    <span
      data-radium={true}
      style={
        Object {
          "marginLeft": 8,
        }
      }
    />
    <i
      aria-hidden={true}
      data-radium={true}
      onClick={undefined}
      style={
        Object {
          "fontFamily": "ic-icons",
          "fontSize": "16px",
          "fontSmoothing": "antialiased",
          "fontStyle": "normal",
          "fontVariant": "normal",
          "fontWeight": "normal",
          "lineHeight": "1",
          "osxFontSmoothing": "grayscale",
          "position": "relative",
          "speak": "none",
          "textTransform": "none",
        }
      }
    >
      
    </i>
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Button renders with inverted colors 1`] = `
<div
  data-radium={true}
>
  <button
    data-radium={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "backgroundImage": "none",
        "border": "1px solid #43B02A",
        "borderRadius": "4px",
        "color": "#43B02A",
        "cursor": "pointer",
        "display": "inline-flex",
        "fontSize": "14px",
        "fontWeight": 600,
        "height": "32px",
        "paddingLeft": 8,
        "paddingRight": 8,
        "touchAction": "manipulation",
        "userSelect": "none",
        "whiteSpace": "nowrap",
      }
    }
    tabIndex={undefined}
    type="button"
  >
    Hi
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
